src/plugins/kibana/server/lib/ingest_build_request.js

```diff
@@ -0,0 +1,24 @@
+const _ = require('lodash');
+const processorTypes = require('../../common/ingest_processor_types');
+
+export default function ingestBuildRequest(pipeline) {
+  const processors = pipeline.processors;
+  const body = {
+    'pipeline': {
+      'processors': []
+    },
+    'docs': [
+      {
+        _source: pipeline.rootObject
+      }
+    ]
+  };
+
+  processors.forEach((processor) => {
+    const processorType = _.find(processorTypes, { 'typeId': processor.typeId });
+    const definition = processorType.getDefinition(processor);
+    body.pipeline.processors.push(definition);
```

Comment:
This loop with throw an error if `processor.typeId` doesn't exist, or the id isn't found in `processorTypes`. Should probably check for that and throw a more useful error that the Hapi request handler can catch and then return a 400 Bad Request.